samplemanager/sample_manager.py

```diff
@@ -304,57 +304,183 @@ def create_production_file(self):
+                "Match main database info in filelist file", 
+                "Pull cross section from other era (only updates the main database)"
+            ],
+            style=custom_style,
+        ).ask()
+        if mode == "Generate filelist file from main database":
+            # check if there are missing detailed database files
+            with Progress() as progress_bar:
+                task = progress_bar.add_task("Samples read ", total=len(self.database.database))
+                for sample in self.database.database:
+                    sampledata = self.database.database[sample]
+                    filelist_json = filelist_path(self.database_folder, sampledata)
+                    if not os.path.isfile(filelist_json):
+                        questionary.print(f"Creating detailed filelist for {sample}")
+                        # first run the query
+                        sampledata["filelist"] = DASQuery(
+                            instance=sampledata["instance"],
+                            nick=sampledata["dbs"],
+                            type="details_with_filelist",
+                            database_folder=self.database_folder,
+                            redirector=self.redirector,
+                        ).result["filelist"]
+                        self.generate_detailed_filelist(sampledata)
+                    else:
+                        pass
+                    progress_bar.update(task, advance=1)
+
+            self.database.database_maintenance()
+
+        elif mode == "Generate datasets entry from filelist file":
+            # check if there are missing entries from the details database
+            with Progress() as progress_bar:
+                task = progress_bar.add_task("Files read ", total=len(self.database.database))
+                datasets = {}
+                for era in os.listdir(self.database_folder):
+                    era_path = os.path.join(self.database_folder, era)
+                    if not os.path.isdir(era_path):
+                        continue
+                    for type in os.listdir(era_path):
+                        type_path =  os.path.join(era_path, type)
+                        if not os.path.isdir(type_path):
+                            continue
+                        for filename in os.listdir(type_path):
+                            if filename.endswith('.json'):
+                                file_path = os.path.join(type_path, filename)
+
+                                with open(file_path, 'r') as f:
+                                    try:
+                                        file_content = json.load(f)
+                                    except json.JSONDecodeError:
+                                        questionary.print(f"Invalid JSON in file: {file_path}")
+                                        continue
+
+                                file_content.pop('filelist', None)
```

Comment:
We are usually using `""` instead of `''`, I suggest to stay consistent with that.